### PR TITLE
Fix import selected groups

### LINKED ISSUE #8933: New entry: not always added to the selected group
### JabRef version

Latest development branch build (please note build date below)

### Operating system

GNU / Linux

### Details on version and operating system

JabRef 5.7--2022-06-27--586d3e4 Linux 4.19.0-20-amd64 amd64  Java 18.0.1  JavaFX 18+12

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

With a group selected:
- when an new entry is created by pasting a DOI in the entry editor, this new entry is added to the group.
- when an new entry is created by clicking on the new entry icon, this new entry is NOT added to the group. 
- 
This is inconsistent. Since the user decided to select a specific group beforehand, I believe the entry should be added to the group in all cases.

To reproduce these 2 behaviors;
1. Open a library with groups (such as https://github.com/JabRef/jabref/blob/main/src/test/resources/testbib/jabref-authors.bib)
2. Select one group
3. Paste a DOI (such as `10.1016/j.jmsy.2021.09.011`)  in the entry editor
==> The entry is added to the selected group
4. Click on the `New article` icon
5. Fill the fields
6. Close the entry editor
==> The entry is not added to the group

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```

</details>

### LINKED ISSUE #11085: Import from bibtex on clipboard to a custom group
Hi, all. In scenarios decribed in the title, I expect the imported entried to appear under the custom group rather than the default "all entries" group. 
But what I found is when there is only one entry on clipboard, it works well. When the number of entries on the clipboard are two or more, only the last one will appear under custom group, the others will appear under "all entries" group.
Please fix or make improvements. When I import multiple entries from zotero using bibtex, I want paste them under custom group. If they get scattered into "all entries" group, it is very time-consuming to find them one-by-one and recollect them to custom group.
Thanks!

### LINKED ISSUE #11269: Selected group loses focus after paste doi
JabRef 5.14--2024-04-30--4f87740
Windows 11 10.0 amd64 
Java 21.0.2 
JavaFX 22.0.1+7

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Select a group
2. Copy a doi from somewhere 
3. Click anywhere in the table and paste, jabref will correctly create a new entry with the selected group assigned
4. Copy another doi from somewhere
5. Click anywhere in the table and paste, jabref will create a new entry but not assigned to the group. In fact it will not show up in "all entries" immediately, and one needs to click a few random groups back and forth, until that works. But still, the new entry is not assigned to the group that was selected first.

Note: 
This is related to issue #8933, it seems that after the first copy-paste in both instances, the group is somehow "deselected", even though it is still shown as selected.
Issue  #11085 might be related too

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #11085 
Fixes #11269
Fixes #8933

Equals/hashcode changes too much

1. Create a new empty library
2. add a subgroup
3. Paste two or more entries in the subgroup
4. 

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We fixed an issue where library shown as unsaved and marked (*) after accepting changes made externally to the file. [#11027](https://github.com/JabRef/jabref/issues/11027)
 - We fixed an issue where drag and dropping entries from one library to another was not always working. [#11254](https://github.com/JabRef/jabref/issues/11254)
 - We fixed an issue where drag and dropping entries created a shallow copy. [#11160](https://github.com/JabRef/jabref/issues/11160)
+- We fixed an issue where imports to a custom group would only worl for the first entry [#11085](https://github.com/JabRef/jabref/issues/11085)
```
Comment (HoussemNasri) [inline CHANGELOG.md:39]: Is one changelog entry enough? There are two issues targeted by this PR and a user reported that it fixed a third issue.

File: src/main/java/org/jabref/model/database/BibDatabaseContext.java [hunk 2]
```diff
@@ -286,4 +293,13 @@ public boolean equals(Object o) {
...
         return Objects.hash(database, metaData, path, location);
     }
+
+    /**
+     * Get the generated UID for the current context. Can be used to distinguish contexts with changing metadata etc
+     *
+     * @return The generated UID in UUIDv4 format
+     */
+    public String getUid() {
+        return uid;
```
Comment (HoussemNasri) [inline src/main/java/org/jabref/model/database/BibDatabaseContext.java:303]: The usage of a plain UUID as a key was confusing while using the debugger, it wasn't intuitive to know what database/library it was referring to:

image


Maybe we do something like `{databasename}_{uid}`?

Comment (Siedlerchr) [reply src/main/java/org/jabref/model/database/BibDatabaseContext.java:303]: I was thinking about such things as well and tried some things, however 
1. A new unsaved database doesn't have a path
2. If it's a shared database it doesn't have a path as well, it has a shared database
3. Saving under a new path/name will change the key, and we lose the reference

However, I now added the prefix bibdatabasecontext to. make it a bit more clear where the uid is related to

Comment (HoussemNasri) [reply src/main/java/org/jabref/model/database/BibDatabaseContext.java:303]: How about? 
```python
if databasename is empty:
     uid = "bibdatabasecontext_${uid}"
else 
    uid = "${databasename}_${uid}"
```

Comment (HoussemNasri) [reply src/main/java/org/jabref/model/database/BibDatabaseContext.java:303]: > Saving under a new path/name will change the key, and we lose the reference

Ah okay I see, I'm fine with the `bibdatabasecontext_` prefix for now but we really should consider other options later.

**Review by HoussemNasri — APPROVED** [review]
LGTM! Sorry for the delay to approve.

**Pull request merged**